plugins/kotlin/jvm-debugger/core/src/org/jetbrains/kotlin/idea/debugger/core/stepping/KotlinRequestHint.kt

```diff
@@ -187,6 +195,15 @@ class KotlinStepIntoRequestHint(
                 lastWasKotlinFakeLineNumber = false
                 return STOP
             }
+
+            // Step over `component*` function calls to let destructured parameters be initialized
+            if (location != null && shouldStepOverParameterDestructuring(context, frameProxy, location)) {
+                // Switching to minimal stepping size allows us to not skip locations of interest
+                // in case of one-line lambdas.
+                size = StepRequest.STEP_MIN
```

Comment:
We have to change the stepping size here, otherwise for cases like `foo { (x, y) -> println("$x$y") }` we will just step on the next line. However changing the stepping size to STEP_MIN in the constructor of `KotlinStepIntoRequestHint` will just slow down the stepping for all cases. Hence the stepping size switching.